docs/usage/dns-autoscaling.md

```diff
@@ -0,0 +1,63 @@
+autoscaling.
+Please note that there is also the option to use a [node-local DNS cache](node-local-dns.md), which helps mitigate potential DNS bottlenecks, but is not discussed in detail in this guide.
+
+## Configuring cluster-proportional DNS Autoscaling
+
+All that needs to be done to enable the usage of cluster-proportional autoscaling of CoreDNS is to set the corresponding option (`spec.systemComponents.coreDNS.autoscaling.mode`) in the `Shoot` resource to `cluster-proportional`:
+
+```yaml
+...
+spec:
+  ...
+  systemComponents:
+    coreDNS:
+      autoscaling:
+        mode: cluster-proportional
+...
+```
+
+To switch back to horizontal DNS autoscaling you can set the `spec.systemComponents.coreDNS.autoscaling.mode` to `horizontal` (or remove the `coreDNS` section).
+
+Once the cluster-proportional autoscaling of CoreDNS has been enabled and the Shoot cluster has been reconciled afterwards, a ConfigMap called `kube-dns-autoscaler` will be created in the `kube-system` namespace with the default settings. The content will be similar to the following:
+
+```yaml
+linear: '{"coresPerReplica":256,"min":2,"nodesPerReplica":16}'
+```
+
+It is possible to adapt the ConfigMap according to your needs in case the defaults do not work as desired. The number of CoreDNS replicas is calculated according to the following formula:
+
+```
+replicas = max( ceil( cores × 1 / coresPerReplica ) , ceil( nodes × 1 / nodesPerReplica ) )
+```
+
+Depending on your needs, you can adjust `coresPerReplica` or `nodesPerReplica`, but it is also possible to override `min` if required.
+
+## Trade-offs of horizontal and cluster-proportional DNS Autoscaling
+
+The horizontal autoscaling of CoreDNS as implemented by Gardener is fully managed, i.e. you do not need to perform any configuration changes. It scales according to the cpu usage of CoreDNS replicas meaning that it will create new replicas if the existing ones are under heavy load. However, it is limited to scale between 2 and 5 instances.
+
+The cluster-proportional autoscaling of CoreDNS as implemented by Gardener is fully managed, but allows more configuration options to adjust the default settings to your individual needs. It scales according to the cluster size, i.e. if your cluster grows in terms of cores/nodes so will the amount of CoreDNS replicas. However, it does not take the actual workload, e.g. cpu consumption, into account.
+
+Experience shows that the horizontal autoscaling of CoreDNS works for a variety of workloads. It does reach its limits if a cluster has a high amount of DNS requests, though. The cluster-proportional autoscaling approach allows to fine-tune the amount of CoreDNS replicas. It helps to scale in clusters of changing size. However, please keep in mind that you need to cater for the maximum amount of DNS requests as the replicas will not be adapted according to the workload, but only according to the cluster size (cores/nodes).
+
+## Trade-offs in conjunction with NodeLocalDNS
```

Comment:
Above you write

> Please note that there is also the option to use a [node-local DNS cache](node-local-dns.md), which helps mitigate potential DNS bottlenecks, but is not discussed in detail in this guide.

but now you start discussing it :) Maybe not in the detail you had in mind when writing the above, however, I would adapt above sentence to not confuse readers.

Reply:
Rephrased to be hopefully less confusing.